diff --git a/src/common/VirtualTerminal.cpp b/src/common/VirtualTerminal.cpp
index 9c0261e..99c4bb9 100644
--- a/src/common/VirtualTerminal.cpp
+++ b/src/common/VirtualTerminal.cpp
@@ -37,8 +37,13 @@
 #include <qscopeguard.h>
 #include <QFileInfo>
 
+#ifdef __FreeBSD__
+#define RELEASE_DISPLAY_SIGNAL (SIGUSR1)
+#define ACQUIRE_DISPLAY_SIGNAL (SIGUSR2)
+#else
 #define RELEASE_DISPLAY_SIGNAL (SIGRTMAX)
 #define ACQUIRE_DISPLAY_SIGNAL (SIGRTMAX - 1)
+#endif
 
 namespace SDDM {
     namespace VirtualTerminal {
@@ -94,6 +99,10 @@ namespace SDDM {
             setModeRequest.mode = VT_PROCESS;
             setModeRequest.relsig = RELEASE_DISPLAY_SIGNAL;
             setModeRequest.acqsig = ACQUIRE_DISPLAY_SIGNAL;
+#ifdef __FreeBSD__
+            // unused but must be set
+            setModeRequest.frsig = SIGINT;
+#endif
 
             if (ioctl(fd, VT_SETMODE, &setModeRequest) < 0) {
                 qDebug() << "Failed to manage VT manually:" << strerror(errno);
diff --git a/src/helper/backend/PamBackend.cpp b/src/helper/backend/PamBackend.cpp
index f371c33..adf85df 100644
--- a/src/helper/backend/PamBackend.cpp
+++ b/src/helper/backend/PamBackend.cpp
@@ -265,6 +265,10 @@ namespace SDDM {
                 m_pam->setItem(PAM_TTY, qPrintable(display));
 #endif
             }
+        } else if (sessionType == QLatin1String("wayland")) {
+                // communicate with pam_ck_connector.so
+                // this is required to pass the check necessary to register this as a graphical session on FreeBSD
+                sessionEnv.insert(QStringLiteral("CKCON_X11_DISPLAY"), QLatin1String("wayland"));
         }
 
         if (!m_pam->putEnv(sessionEnv)) {
@@ -275,6 +279,7 @@ namespace SDDM {
             m_app->error(m_pam->errorString(), Auth::ERROR_INTERNAL);
             return false;
         }
+        sessionEnv.remove(QStringLiteral("CKCON_X11_DISPLAY"));
         sessionEnv.insert(m_pam->getEnv());
         m_app->session()->setProcessEnvironment(sessionEnv);
         return Backend::openSession();
