--- a/configure.ac	2024-06-07 09:17:50.608162031 -0700
+++ b/configure.ac	2024-06-07 09:45:59.844518241 -0700
@@ -339,6 +339,9 @@
 	*-*-wasi)
 	    ac_sys_system=WASI
 	    ;;
+	*-*-freebsd)
+	    ac_sys_system=FreeBSD
+	    ;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
@@ -777,6 +780,9 @@
 	wasm32-*-* | wasm64-*-*)
 		_host_ident=$host_cpu
 		;;
+	*-*-freebsd)
+		_host_ident=$host_cpu
+		;;
 	*)
 		# for now, limit cross builds to known configurations
 		MACHDEP="unknown"
